src/Bicep.Core/Emit/ParametersJsonWriter.cs

```diff
@@ -36,16 +36,43 @@ public JToken GenerateTemplate()
         jsonWriter.WritePropertyName("parameters");
         jsonWriter.WriteStartObject();
 
-        foreach (var parameter in model.Root.ParameterAssignments)
+        foreach (var assignment in model.Root.ParameterAssignments)
         {
-            jsonWriter.WritePropertyName(parameter.Name);
+            jsonWriter.WritePropertyName(assignment.Name);
+
+            var parameter = model.EmitLimitationInfo.ParameterAssignments[assignment];
 
-            jsonWriter.WriteStartObject();
+            if (parameter.Type == JTokenType.Object)
+            {
+                var propertyName = (parameter.First as JProperty)?.Name;
+                if (propertyName != "reference")
+                {
```

Comment:
It's likely this check will have some false positives when users supply an object literal that just happens to contain a `reference` property. For example, I would expect the following:

```bicep
param foo = {
  reference: 'bar'
}
```

to compile to:

```json
"foo": {
  "value": {
    "reference": "bar"
  }
}
```

but it would instead compile to:

```json
"foo": {
  "reference": "bar"
}
```

which would cause the deployment to fail.

We may need to introduce a new IR expression type in order to be able to return something recognizable from `az.getSecret(...)`. `model.EmitLimitationInfo.ParameterAssignments`'s would also need to be modified to either have a constant value **or** a KV reference.

Reply:
Here's how we detect and handle `getSecret` in regular `.bicep` code-gen: https://github.com/Azure/bicep/blob/eb4486b4a5370cfc531df72618e4e9f749a3aa6e/src/Bicep.Core/Emit/ExpressionEmitter.cs#L331-L332

I'd recommend taking a similar approach if possible

Reply:
> ```bicep
> param foo = {
>   reference: 'bar'
> }
> ```

Could you add the above as a test case? I don't quite follow how the change in d5a685b4d3dd930e5eca7bedb3e4c958836ca125 detects that this is a value rather than a reference.

Reply:
created a new commit with the added test case, all the tests are passing 👍